components/eventing-controller/testing/matchers.go

```diff
@@ -0,0 +1,181 @@
+package testing
+
+import (
+	"log"
+	"net/http"
+	"net/url"
+	"reflect"
+	"strings"
+
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gstruct"
+	. "github.com/onsi/gomega/types"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	apigatewayv1alpha1 "github.com/kyma-incubator/api-gateway/api/v1alpha1"
+	eventingv1alpha1 "github.com/kyma-project/kyma/components/eventing-controller/api/v1alpha1"
+	"github.com/kyma-project/kyma/components/eventing-controller/pkg/constants"
+	"github.com/kyma-project/kyma/components/eventing-controller/pkg/object"
+	"github.com/kyma-project/kyma/components/eventing-controller/utils"
+)
+
+func HaveSubscriptionName(name string) GomegaMatcher {
+	return WithTransform(func(s eventingv1alpha1.Subscription) string { return s.Name }, Equal(name))
+}
+
+func HaveSubscriptionFinalizer(finalizer string) GomegaMatcher {
+	return WithTransform(func(s eventingv1alpha1.Subscription) []string { return s.ObjectMeta.Finalizers }, ContainElement(finalizer))
+}
+
+func IsAnEmptySubscription() GomegaMatcher {
+	return WithTransform(func(s eventingv1alpha1.Subscription) bool {
+		emptySub := eventingv1alpha1.Subscription{}
+		if reflect.DeepEqual(s, emptySub) {
+			return true
+		}
+		return false
+	}, BeTrue())
+}
+
+func IsAnEmptyAPIRule() GomegaMatcher {
+	return WithTransform(func(a apigatewayv1alpha1.APIRule) bool {
+		if a.Name == "" {
+			return true
+		}
+		return false
+	}, BeTrue())
+}
+
+func HaveSubscriptionReady() GomegaMatcher {
+	return WithTransform(func(s eventingv1alpha1.Subscription) bool {
+		return s.Status.Ready
+	}, BeTrue())
+}
+
+func HaveSink(sink string) GomegaMatcher {
+	return WithTransform(func(s eventingv1alpha1.Subscription) bool {
+		return s.Spec.Sink == sink
+	}, BeTrue())
+}
+
+func HaveApiRuleReady() GomegaMatcher {
+	return WithTransform(func(a apigatewayv1alpha1.APIRule) bool {
+		if a.Status.APIRuleStatus == nil || a.Status.AccessRuleStatus == nil || a.Status.VirtualServiceStatus == nil {
+			return false
+		}
+		apiRuleStatus := a.Status.APIRuleStatus.Code == apigatewayv1alpha1.StatusOK
+		accessRuleStatus := a.Status.AccessRuleStatus.Code == apigatewayv1alpha1.StatusOK
+		virtualServiceStatus := a.Status.VirtualServiceStatus.Code == apigatewayv1alpha1.StatusOK
+		return apiRuleStatus && accessRuleStatus && virtualServiceStatus
+	}, BeTrue())
+}
+
+func HaveValidAPIRule(s *eventingv1alpha1.Subscription) GomegaMatcher {
+	return WithTransform(func(apiRule apigatewayv1alpha1.APIRule) bool {
+		hasOwnRef, hasRule := false, false
```

Comment:
During the tests, WithTransform() is called with the parameter apiRule as a "pointer" to "apigatewayv1alpha1.APIRule". This code snipped solv it:

```
return WithTransform(func(apiRule *apigatewayv1alpha1.APIRule) bool {
    if apiRule == nil {
        log.Printf("!!!!!! apiRule == nill")
	return false
    }
```